applications/tari_validator_node/src/p2p/services/template_manager/service.rs

```diff
@@ -168,14 +181,22 @@ impl TemplateManagerService {
     async fn handle_add_template(&mut self, template: TemplateRegistration) -> Result<(), TemplateManagerError> {
         let address = template.template_address;
         let url = template.registration.binary_url.to_string();
+        let expected_binary_sha = template.registration.binary_sha.clone();
```

Comment:
I think we should convert this to a hash, the binary_sha is wrong (my bad :smile: ) should be binary_hash and be a FixedHash type. But for now, we can use it as a hash in the DAN layer:
```suggestion
        let expected_binary_hash = FixedHash::try_from(template.registration.binary_sha.into_vec())?;
```